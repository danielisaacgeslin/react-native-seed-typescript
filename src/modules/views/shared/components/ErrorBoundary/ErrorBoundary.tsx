import React from 'react';
import { Text } from 'react-native';
import { Logger } from '../../../../services/Logger';

export interface IErrorBoundaryState {
  hasError: boolean;
}

export default class ErrorBoundary extends React.PureComponent<{}, IErrorBoundaryState> {
  private static defaultProps = {
    children: null
  };
  public state = { hasError: false };

  public componentDidCatch(error, info) {
    // eslint-disable-line
    Logger.error('ErrorBoundary: ', error, info);
    this.setState({ hasError: true });
  }

  public render() {
    if (this.state.hasError) return <Text>Error was catched</Text>;
    else return this.props.children; // eslint-disable-line
  }
}
