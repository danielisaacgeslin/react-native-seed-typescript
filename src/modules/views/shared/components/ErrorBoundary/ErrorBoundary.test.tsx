import React from 'react';

import ErrorBoundary from './ErrorBoundary';
import renderer from 'react-test-renderer';
import { Text } from 'react-native';

class HandGranade extends React.Component {
  public render() {
    const a: any = null;
    a.b.c = 10;
    return <Text>This should not work</Text>;
  }
}

describe('ErrorBoundaryTest', () => {
  global.console.error = () => {
    /** */
  };
  it('should render child', () => {
    const rendered = renderer
      .create(
        <ErrorBoundary>
          <Text>Render test</Text>
        </ErrorBoundary>
      )
      .toJSON();
    expect(rendered).toMatchSnapshot();
  });

  it('should render placeholder when child render fails', () => {
    const rendered = renderer
      .create(
        <ErrorBoundary>
          <HandGranade>Render test</HandGranade>
        </ErrorBoundary>
      )
      .toJSON();
    expect(rendered).toMatchSnapshot();
  });
});
