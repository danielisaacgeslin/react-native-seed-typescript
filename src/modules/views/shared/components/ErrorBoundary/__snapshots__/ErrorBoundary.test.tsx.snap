// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundaryTest should render child 1`] = `
<Text>
  Render test
</Text>
`;

exports[`ErrorBoundaryTest should render placeholder when child render fails 1`] = `
<Text>
  Error was catched
</Text>
`;
